src/cucumber-runner.ts

```diff
@@ -67,6 +71,43 @@ function buildArgs(runCfg: CucumberRunnerConfig, cucumberBin: string) {
   return procArgs;
 }
 
+/**
+ * Normalizes a Cucumber-js format string.
+ *
+ * For structured inputs (`key:value` or `"key:value"`), returns a string in the
+ * form `"key":"value"`, with the asset directory prepended to relative paths.
+ * For simple inputs (e.g., `usage`), returns the input as-is.
+ *
+ * @param {string} format - The input format string. Examples include:
+ *                          - `"key:value"`
+ *                          - `"key":"value"`
+ *                          - `key:value`
+ *                          - `usage`
+ * @param {string} assetDir - The directory to prepend to the value for relative paths.
+ * @returns {string} The normalized format string. For structured inputs, it returns
+ *                   a string in the form `"key":"value"`. For simple inputs, it
+ *                   returns the input unchanged.
+ *
+ * Example:
+ * - Input: `"html":"formatter/report.html"`, `"/project/assets"`
+ *   Output: `"html":"/project/assets/formatter/report.html"`
+ * - Input: `"usage"`, `"/project/assets"`
+ *   Output: `"usage"`
+ */
+export function normalizeFormat(format: string, assetDir: string): string {
+  // Checks if the format is structured; if not, returns it unchanged.
+  const match = format.match(/^"?([^:]+):"?([^"]+)"?$/);
```

Comment:
This is a valid formatter that, I think, will get you a match, but the groups will incorrectly be `file` and `://path/to/customFormatter`:

`file://path/to/customFormatter`

The valid form for formatters, from the [docs](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md):

* The name of one of the built-in formatters (below) e.g. progress-bar
* A module/package name e.g. @cucumber/pretty-formatter
* A relative path to a local formatter implementation e.g. ./my-custom-formatter.js
* An absolute path to a local formatter implementation in the form of a file:// URL

Would be good if your unit test covered these cases.

Reply:
Good point.